Configure workload access outside cluster (#2908)
projectcalico/calico: master/networking/workloads-outside-cluster.md

---
title: Configure workload access outside the cluster
---

### Big picture

With all {{site.prodname}} networking options, you can enable and disable internet access for your workloads.

### Value

You control which {{site.prodname}} IP pools are subject to outbound NAT to enable public and private network connections and allow single IP address communication. 

### Features

This how-to guide uses the following {{site.prodname}} features:

- **IPPool** resource with natOutgoing field

### Concepts

#### Outgoing NAT on each node

> For all networking options, you can allow your workloads to access the internet (subject to network policy, of course!) by enabling outbound NAT on the IP pool.  When enabled, traffic from pods in that pool go to a destination outside all {{site.prodname}} pools will be NATed.  The source address is masqueraded to an IP of the node on which each workload is hosted, allowing the network to understand the traffic. By default, the NAT outgoing option is automatically enabled for the pool that is created when you install {{site.prodname}}.  

Reviewer: ```suggestion
For all networking options, you can allow your workloads to access the internet (subject to network policy, of course!) by enabling outbound NAT on the IP pool.  When enabled, traffic from pods in that pool going to a destination outside all {{site.prodname}} pools will be NATed.  The source address is masqueraded to an IP of the node on which each workload is hosted, allowing the network to understand the traffic. By default, the NAT outgoing option is automatically enabled for the pool that is created when you install {{site.prodname}}.  
```

Reviewer: May be hard to see but the change above is:
```traffic from pods in that pool go to a destination outside all {{site.prodname}} pools will be NATed```
to
```traffic from pods in that pool going to a destination outside all {{site.prodname}} pools will be NATed```

#### BGP peering with out-of-cluster NAT

When {{site.prodname}} is [BGP peered with your physical network infrastructure]({{site.baseurl}}/{{page.version}}/networking/bgp), you can use the infrastructure to NAT traffic from pods to the internet.  In this case, you should disable the {{site.prodname}} NAT outgoing option. 

#### Expose pods directly on the internet

If you want your pods to have public internet IPs:

> - Configure {{site.prodname}}to peer with your physical network infrastructure

Reviewer: ```suggestion
- Configure {{site.prodname}} to peer with your physical network infrastructure
```

> - Create an IP pool for those pods that contain public IP addresses that are routed to your network with `nat-outgoing: false`

Reviewer: ```suggestion
- Create an IP pool for those pods that contain public IP addresses that are routed to your network with `natOutgoing: false`
```
